xla/service/elemental_ir_emitter.cc

```diff
@@ -220,6 +220,59 @@ absl::StatusOr<llvm::Value*> EmitReducePrecisionIR(
   return result;
 }
 
+llvm::Value* handle_halfway_points_F16ToF8(llvm::Value* f16_abs_bits,
```

Comment:
Also, I think you might have to modify [expand_float_ops.cc](https://github.com/openxla/xla/blob/main/xla/service/gpu/fusions/transforms/expand_float_ops.cc), which is used by the new MLIR emitters which replace the existing emitters on GPUs. But I'm not very familiar with these new emitters. @jreiffers, can you advice on what needs to be done here

Reply:
We have generic support for all possible float conversions, but the emitted code might not be optimal, so it should be considered a fallback. I didn't look at these conversion routines here in detail, but if they're better, it would make sense to port them to the MLIR pipeline.

Reply:
I updated `xla/service/gpu/fusions/transforms/expand_float_ops.cc` and added `f8E4M3` cases to:
- IsInf()
- IsNaN()
- RewriteF8Cst::matchAndRewrite() // If we're comparing to +-0, compare the absolute values.

`expand_float_ops.cc` includes a specialized function for the f8e5m2 type - `EmitF16ToF8e5m2()`. This is because F16 is technically `f16e5m10`. The two types are similar, with the primary difference being that the mantissa in `f8e5m2` is truncated to 2 bits.

`f8E4M3` has a different number of exponent and mantissa bits. The conversion can be efficiently managed using the _"generic support for all possible float conversions"._

Tested xla on CUDA:
```bash
//xla/tests/...    799 tests: 799 tests pass
//xla/service/...  865 tests: 865 tests pass
//xla/client/...    77 tests: 77 tests pass
//xla/runtime/...    1 test: 1 test passes
//xla/ffi/...        6 tests: 6 tests pass
//xla/hlo/...       12 tests: 12 tests pass
//xla/mlir/...     141 tests: 141 tests pass
//xla/mlir_hlo/...  98 tests: 98 tests pass
//xla/pjrt/...      26 tests: 26 tests pass
//xla/tools/...     41 tests: 41 tests pass
//xla/translate/... 61 tests: 61 tests pass
```